### PR TITLE
Improve item-number parsing and add mark-read command

### PR DESCRIPTION
**Pull request by jameshaydon**
- Removes duplicate code for checking `number?` and `integral?`.
- Adds checks to some places that were missing.
- Improves checking (catching `read` exceptions).
- Adds `mark-read` cmd to issues and patches.
- Makes action descriptions in some errors more specific.

### CODE REVIEW

File: bin/rad-patch [hunk 1]
```diff
@@ -386,19 +371,24 @@
...
-          (/cmd-1 "accept" 'patch-number help) (edit-patch-state! (machine-name) (read patch-number) :accepted)
-          (/cmd-1 "reject" 'patch-number help) (edit-patch-state! (machine-name) (read patch-number) :rejected)
-          (/cmd-2 "comment" 'patch-number 'comment help) (create-comment! (machine-name) (read patch-number) comment)
+          (/cmd-1 "checkout" 'n help)  (whole-patch-num "checkout" n (fn [n] (checkout-patch! (machine-name) n)))
+          (/cmd-1 "show" 'n help)      (whole-patch-num "show" n (fn [n] (show-patch! (machine-name) n)))
+          (/cmd-1 "mark-read" 'n help) (whole-patch-num "mark as read" n (fn [n] (newness/mark-read! (machine-name) n)))
+          (/cmd-1 "retract" 'n help)   (whole-patch-num "retract" n (fn [n] (edit-patch-state! (machine-name) n :retracted)))
+          (/cmd-1 "accept" 'n help)    (whole-patch-num "accept" n (fn [n] (edit-patch-state! (machine-name) n :accepted)))
+          (/cmd-1 "reject" 'n help)    (whole-patch-num "reject" n (fn [n] (edit-patch-state! (machine-name) n :rejected)))
+          (/cmd-2 "comment" 'n 'comment help) (whole-patch-num "comment on" (fn [n] (create-comment! (machine-name) n comment)))
```
Comment (MeBrei) [inline bin/rad-patch]: n for `num-str` is missing for comment command

File: bin/rad-issue [hunk 2]
```diff
@@ -236,44 +236,29 @@
...
-                 (put-str! (error/state-change-failure :issue "closed"))
-                 (exit! 1)))
-        (cmd-parse-failure (error/whole-item-number :issue)))
-      (cmd-parse-failure (error/missing-item-number :issue "close")))))
+    (catch 'daemon-error
+        (do
+          (verify-issue-number n machine)
+          (simple-edit-issue! machine n {:state :closed})
+          (put-str! (string-append "Issue #" (show n) " has been closed."))
+          (newness/mark-read! machine n))
```
Comment (MeBrei) [inline bin/rad-issue]: indentation

File: bin/rad-issue [hunk 3]
```diff
@@ -236,44 +236,29 @@
...
-                 (put-str! (string-append "Issue #" (show n) " has been closed."))
-                 (newness/mark-read! machine n))
-               (fn [_]
-                 (put-str! (error/state-change-failure :issue "closed"))
-                 (exit! 1)))
-        (cmd-parse-failure (error/whole-item-number :issue)))
-      (cmd-parse-failure (error/missing-item-number :issue "close")))))
+    (catch 'daemon-error
+        (do
+          (verify-issue-number n machine)
```
Comment (MeBrei) [inline bin/rad-issue]: needs to be `issue-number` as `(def n...)` was removed

File: rad/prelude/cmd-parsing.rad [hunk 4]
```diff
@@ -167,3 +167,16 @@
...
+  the integer is passed to handler function. If not, an error is show to the
+  user."
+  (fn [item action num-str f]
+    (def bad (fn [] (cmd-parse-failure (error/missing-item-number item action))))
+    (def n (catch 'any (read num-str) (fn [_] (bad))))
+    (if (number? n)
+      (if (integral? n)
+        (f n)
+        (not-whole))
+      (cmd-parse-failure (error/missing-item-number item action)))))
```
Comment (MeBrei) [inline rad/prelude/cmd-parsing.rad]: Add documentation and testing.

Comment (jameshaydon) [reply rad/prelude/cmd-parsing.rad]: What's wrong with the docs?

Comment (MeBrei) [reply rad/prelude/cmd-parsing.rad]: Sorry, somehow missed the docs.

**Pull request merged**